compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -2457,6 +2457,20 @@ public BLangNode transform(ExpressionStatementNode expressionStatement) {
         switch (kind) {
             case ASYNC_SEND_ACTION:
                 return expressionStatement.expression().apply(this);
+            case WAIT_ACTION:
+                WaitActionNode waitActionNode =
+                        (WaitActionNode) expressionStatement.expression();
+                if (waitActionNode.waitFutureExpr().kind() == SyntaxKind.START_ACTION) {
+                    BLangWorkerSend bLangWorkerSend =
+                            (BLangWorkerSend) TreeBuilder.createWorkerSendNode();
+                    bLangWorkerSend.expr = createExpression(expressionStatement.expression());
+                    bLangWorkerSend.pos = getPosition(expressionStatement);
+                    AsyncSendActionNode asyncSendActionNode =
+                            (AsyncSendActionNode) ((StartActionNode) waitActionNode.waitFutureExpr()).expression();
+                    bLangWorkerSend.setWorkerName(createIdentifier(asyncSendActionNode.peerWorker().name()));
+                    return bLangWorkerSend;
+                }
```

Comment:
Any clue on why do we need to special-case start-acton here?

Reply:
no Idea, that's just the way the node is getting transformed. it doesn't follow that path for anything else.

Reply:
no i meant, what was the reason to special case the start-action?

Reply:
well when we have a start action inside a wait action the old parser doesnt return an expressionstmt node but rather a workersendNode. so thats the reason

Reply:
even when the original node is an expressionstatement node